parser/src/test/java/com/facebook/coresql/parser/sqllogictest/java/SqlLogicTest.java

```diff
@@ -0,0 +1,216 @@
+import java.nio.file.Paths;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.HashSet;
+import java.util.Set;
+
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTest.AcceptedRdbms.ALL;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestConfig.TEST_FILES_ROOT_PATH;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.createFileForResults;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.validateStatementAgainstPossibleErrors;
+import static com.facebook.coresql.parser.sqllogictest.java.SqlLogicTestUtils.writeTestResultToFile;
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static java.nio.file.Files.newBufferedReader;
+import static java.nio.file.Files.walkFileTree;
+
+public final class SqlLogicTest
+{
+    private SqlLogicTestConfig config;
+    private SqlLogicTestResult result;
+
+    private Set<AcceptedRdbms> skipIfSet;
+    private AcceptedRdbms onlyIf;
+    private FileVisitor<Path> fileVisitor;
+
+    protected enum SqlLogicTestError
+    {
+        PARSING_ERROR,
+        UNPARSING_ERROR,
+        UNPARSED_DOES_NOT_MATCH_ORIGINAL_ERROR
+    }
+
+    protected enum AcceptedRdbms
+    {
+        ALL,
+        ORACLE,
+        MYSQL,
+        MSSQL,
+        SQLITE,
+        POSTGRESQL
+    }
+
+    public SqlLogicTest() {}
+
+    public void run()
+            throws IOException
+    {
+        setup();
+        walkFileTree(Paths.get(TEST_FILES_ROOT_PATH), fileVisitor);
+        writeTestResultToFile(result);
+    }
+
+    private void setup()
+            throws IOException
+    {
+        config = new SqlLogicTestConfig();
+        result = new SqlLogicTestResult(config.getRdbms());
+        fileVisitor = new SqlLogicTestFileVisitor();
+        skipIfSet = new HashSet<>();
+        createFileForResults();
+    }
+
+    private void validateParsingOfSqlStatement(String statement)
+    {
+        SqlLogicTestError error = validateStatementAgainstPossibleErrors(statement);
+        if (error == null) {
+            result.incrementNumStatementsParsed();
+        }
+        else {
+            result.addErroneousStatement(new ErroneousSqlStatement(statement, error));
+        }
+    }
+
+    private class SqlLogicTestFileVisitor
+            implements FileVisitor<Path>
+    {
+        private void processStatement(BufferedReader reader)
+                throws IOException
+        {
+            StringBuilder statement = new StringBuilder();
+            while (true) {
+                final String line = reader.readLine();
+                if (line == null || isLineAfterStatement(line)) {
+                    break;
+                }
+                statement.append(line);
+            }
+            if (statement.length() > 0 && !statement.substring(statement.length() - 1).equals(";")) {
+                statement.append(";"); // NOTE: This semi-colon is necessary for parsing.
```

Comment:
nit: remove "NOTE: " A line comment is by definition a note.

Reply:
Done.